### PR TITLE
[NewIR]support c_allreduce_sum/c_identity/c_embedding/c_embedding_grad

### PR DESCRIPTION
**Pull request by zhaoyinglia**
### PR types

New features
### PR changes

OPs
### Description

Pcard-67164
- NewIR support c_allreduce_sum/c_identity/c_embedding/c_embedding_grad for model parallel

### CODE REVIEW

File: paddle/phi/infermeta/binary.h [hunk 1]
```diff
@@ -211,6 +211,11 @@ void EmbeddingInferMeta(const MetaTensor& x,
                         int64_t padding_idx,
                         MetaTensor* out);
 
+void CEmbeddingInferMeta(const MetaTensor& weight,
+                         const MetaTensor& x,
+                         int64_t start_index,
+                         MetaTensor* out);
+
```
Comment (zyfncg) [inline paddle/phi/infermeta/binary.h:218]: InferMeta函数需要按照字典序放置，方便查找

**Review by zyfncg — APPROVED** [review]

File: paddle/phi/infermeta/binary.h (hunk 1 above)
Comment (zhaoyinglia) [reply paddle/phi/infermeta/binary.h:218]: Ok, will fix it in the next pr.

**Review by lanxianghit — APPROVED** [review]
LGTM for new c++ ops

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs，no docs changes

**Pull request merged**